crates/astria-sequencer/src/bridge/bridge_lock_action.rs

```diff
@@ -319,45 +312,63 @@ mod tests {
         let calculated_len_3 = get_deposit_byte_len(&deposit);
         assert!(calculated_len_3 >= calculated_len_2);
 
-        // Ensure calculated length is still greater than or equal to actual with absurd string
+        // Ensure calculated length is as expected with absurd string
         // lengths (have tested up to 99999999, but this makes testing very slow)
         let absurd_string: String = ['a'; u16::MAX as usize].iter().collect();
         let deposit = Deposit::new(
             astria_address(&[1; 20]),
             RollupId::from_unhashed_bytes(b"test_rollup_id"),
             u128::MAX,
             absurd_string.parse().unwrap(),
-            absurd_string,
+            absurd_string.clone(),
             TransactionId::new([0; 32]),
             u64::MAX,
         );
         let calculated_len = get_deposit_byte_len(&deposit);
-        let expected_len = deposit.into_raw().encoded_len() as u128;
-        assert!(calculated_len >= expected_len);
+        let expected_len = 16 + u128::from(u16::MAX) * 2;
+        assert_eq!(calculated_len, expected_len);
+    }
+
+    #[test]
+    fn get_deposit_byte_length_snapshot() {
+        let deposit = Deposit::new(
+            astria_address(&[1; 20]),
+            RollupId::from_unhashed_bytes(b"test_rollup_id"),
+            u128::MAX,
+            test_asset(),
+            "someaddress".to_string(),
+            TransactionId::new([0; 32]),
+            u64::MAX,
+        );
+        assert_json_snapshot!(get_deposit_byte_len(&deposit));
```

Comment:
To be honest I am not a fan of making this a snapshot test. We must provide a test for calculating `DEPOSIT_BASE_BYTE_LENGTH` in code.

I guess we can also add a snapshot test just for `DEPOSIT_BASE_BYTE_LENGTH` alone even though the use of that is questionable.